awx/sso/backends.py

```diff
@@ -423,3 +457,28 @@ def on_populate_user(sender, **kwargs):
     if profile.ldap_dn != ldap_user.dn:
         profile.ldap_dn = ldap_user.dn
         profile.save()
+
+    # Get all of the orgs listed in LDAP from the database
+    roles = ['admin_role', 'auditor_role', 'member_role']
+    prefetch_roles = ["{}__members".format(role_name) for role_name in roles]
+    orgs_tied_to_ldap = Organization.objects.filter(name__in=desired_org_states.keys()).prefetch_related(*prefecth_roles)
+    # Set the state from memory into the org objects
+    for org in orgs_tied_to_ldap:
+        for role in roles:
+            if desired_org_states[org.name][role] is None:
+                # If something got a none we just need to continue on
+                pass
+            elif desired_org_states[org.name][role]:
+                getattr(org, role).members.add(user)
+            elif user in getattr(org, role).members.all():
+                getattr(org, role).members.remove(user)
+
+    # Get all of the teams listed in LDAP from the database
+    teams_tied_to_ldap = Team.objects.filter(name__in=desired_team_states.keys()).prefetch_related('member_role__members')
```

Comment:
same thing here, doing the prefetch saves us a bunch of queries on ` team.member_role.members`